routing/cross_mwm_ramp.cpp

```diff
@@ -0,0 +1,121 @@
+#include "routing/cross_mwm_ramp.hpp"
+
+namespace
+{
+uint32_t constexpr kFakeId = std::numeric_limits<uint32_t>::max();
+}  // namespace
+
+namespace routing
+{
+// static
+CrossMwmRamp::Weight constexpr CrossMwmRamp::kNoRoute;
+
+void CrossMwmRamp::AddTransition(uint32_t featureId, uint32_t segmentIdx, bool oneWay,
+                                 bool forwardIsEnter, m2::PointD const & backPoint,
+                                 m2::PointD const & frontPoint)
+{
+  Transition transition(kFakeId, kFakeId, oneWay, forwardIsEnter, backPoint, frontPoint);
+
+  if (forwardIsEnter)
+  {
+    transition.m_enterIdx = base::asserted_cast<uint32_t>(m_enters.size());
+    m_enters.emplace_back(m_mwmId, featureId, segmentIdx, true);
```

Comment:
So, from my point of view, num mwm ids can be used only during the application lifetime, not between consecutive application runs even on the same device.